app/models/manageiq/providers/azure/cloud_manager/provision_workflow.rb

```diff
@@ -0,0 +1,45 @@
+class ManageIQ::Providers::Azure::CloudManager::ProvisionWorkflow < ManageIQ::Providers::CloudManager::ProvisionWorkflow
+  def allowed_instance_types(_options = {})
+    source = load_ar_obj(get_source_vm)
+    ems    = source.try(:ext_management_system)
+    return {} if ems.nil?
+    flavors = ems.flavors
+    flavors.each_with_object({}) { |f, hash| hash[f.id] = display_name_for_name_description(f) }
+  end
+
+  def allowed_resource_groups(_options = {})
+    source = load_ar_obj(get_source_vm)
+    ems    = source.try(:ext_management_system)
+    return {} if ems.nil?
+    resource_groups = ems.resource_groups
+    resource_groups.each_with_object({}) { |rg, hash| hash[rg.id] = rg.name }
+  end
+
+  def allowed_cloud_subnets(_options = {})
+    src = resources_for_ui
+    if (cn = CloudNetwork.find_by(:id => src[:cloud_network_id]))
+      cn.cloud_subnets.each_with_object({}) do |cs, hash|
+        hash[cs.id] = "#{cs.name} (#{cs.cidr})"
+      end
+    else
+      {}
+    end
+  end
+
+  def validate_cloud_subnet(field, values, dlg, fld, value)
```

Comment:
This method is the same as the one defined in [amazon/cloud_manager/provision_workflow.rb:L50](https://github.com/ManageIQ/manageiq/blob/master/app/models/manageiq/providers/amazon/cloud_manager/provision_workflow.rb#L50).  You could DRY this method by moving them both to the command parent model `ManageIQ::Providers::CloudManager::ProvisionWorkflow`.